### DISCUSSION TITLE
RadioButton onChange value must be Boolean

### DISCUSSION THREAD

**Discussion by chrishj59** [primefaces/community#160] [PrimeReact] [answered]
Hi,
The onChange event passes a parameter of the type (event:RadioButtonChangeEvent)](https://primereact.org/radiobutton/#api.RadioButton.RadioButtonChangeEvent). This expects a  Boolean, whereas the Radion button may have a value of a string.

Is it possible to change the documentation to use the type any or the  RadioButtonChangeEvent.value to accept the type of the radio button value?

This issue is shown in the Sandbox from the documentation.
Edit: In the Apollo example code, the RadioButton onChange value is sent as without a type

**Comment by chrishj59** [comment]
I believe the issue in radiobutton.d.ts
`interface RadioButtonChangeEvent extends FormBooleanEvent {} `

**Reply by habubey** [reply]
Hey there!, [Radiobutton typescript FormEvent](https://github.com/primefaces/primereact/commit/02ffa72f7c8f88f76121d1c76bbd3063d5ea4317) Do you receive an error in the current state and requires modification?

**Reply by chrishj59** [reply]
I get a typescript error if I use RadioButtonChangeEvent as the type for OnChange.

**Reply by chrishj59** [reply]
If I update radiobutton.d.ts and change the boolean formEvent as below on the local code, then it works okay. I am not sure if this will cause issues elsewhere.
`interface RadioButtonChangeEvent extends FormEvent {}`

The issue with the local change is that when I upgrade to the next version my app will break

**Reply by habubey** [reply]
The current code is like this in radiobutton.d.ts
`interface RadioButtonChangeEvent extends FormEvent {}` if it works for you, It will be fixed in the next release

**Reply by melloware** [reply, marked as answer]
@chrishj59 i already fixed this for 10.0.3: https://github.com/primefaces/primereact/pull/4989  It hasn't been released yet
